apps/block_scout_web/assets/js/omni.js

```diff
@@ -1,199 +1,318 @@
+    const data = await response.json()
     if (data.length === 0) {
-      return undefined;
+      return undefined
     }
-    return data;
+    return data
   } catch (err) {
-    console.error("error getting xchain tx details: ", omniTxHash, " ", err);
-    return undefined;
+    console.error('error getting xchain tx details: ', omniTxHash, ' ', err)
+    return undefined
   }
-};
+}
 
 const getAllTxHashElements = () => {
-  return document.querySelectorAll('[data-test="transaction_hash_link"]');
-};
+  return document.querySelectorAll('[data-test="transaction_hash_link"]')
+}
+
+const addXChainMarket = e => {
+  const marker = xchainMarker()
+  e.appendChild(marker)
+
+  // im sorry
+  // this colors the whole tx block
+  marker.parentElement.parentElement.parentElement.parentElement.parentElement.style.backgroundColor =
+    '#F5F7FE'
+}
 
 const handletxHashLinks = async () => {
-  const elements = getAllTxHashElements();
-  if (elements.length == 0) {
-    return;
+  const elements = getAllTxHashElements()
+  if (elements.length === 0) {
+    return
   }
   // check local cache
-  const unknownTxList = [];
-  for (let i of elements) {
+  const unknownTxList = []
+  for (const i of elements) {
     if (
       i.parentNode.childElementCount <= 2 &&
       txMapCache[i.innerHTML] === true
     ) {
-      console.log("from cache " + i.innerHTML);
-      i.parentNode.appendChild(xchainMarker());
+      addXChainMarket(i.parentNode)
     } else if (txMapCache[i.innerHTML] === undefined) {
-      unknownTxList.push(i.innerHTML);
+      unknownTxList.push(i.innerHTML)
     }
   }
-  if (unknownTxList.length == 0) {
-    return;
+
+  if (unknownTxList.length === 0) {
+    return
   }
-  const xChainTxs = await getXChainTxs(unknownTxList);
+
+  const xChainTxs = await getXChainTxs(unknownTxList)
+
   if (xChainTxs === undefined) {
-    return;
+    return
   }
-  for (let i of elements) {
+
+  for (const i of elements) {
     if (xChainTxs[i.innerHTML] === true) {
-      txMapCache[i.innerHTML] = true;
-      console.log("from api " + i.innerHTML);
-      i.parentNode.appendChild(xchainMarker());
-    } else if (txMapCache[i.innerHTML] == undefined) {
-      txMapCache[i.innerHTML] = false;
+      txMapCache[i.innerHTML] = true
+      addXChainMarket(i.parentNode)
+    } else if (txMapCache[i.innerHTML] == null) {
+      txMapCache[i.innerHTML] = false
     }
   }
-};
+}
 
-const displayTxDetails = async (txHash) => {
+const displayTxDetails = async txHash => {
   const elements = document.querySelectorAll(
-    '[class="card js-ad-dependant-mb-3"]'
-  );
-  if (elements.length != 1) {
-    console.log("could not find unique tx view element");
-    return;
+    '[class="card js-ad-dependant-mb-3"]',
+  )
+  if (elements.length !== 1) {
+    console.log('could not find unique tx view element')
+    return
   }
+  const container = elements[0]
+
+  const details = await getTxDetails(txHash)
+
+  console.log(details)
```

Comment:
do we want to log to console when we merge?

Reply:
no I'll move